subsys/shell/shell_uart.c

```diff
@@ -81,7 +81,9 @@ static void uart_rx_handle(struct device *dev,
 			/* Divert this byte from shell handling if it
 			 * is part of an mcumgr frame.
 			 */
-			smp_shell_rx_byte(&sh_uart->ctrl_blk->smp, dummy);
+			if (smp_shell_rx_byte(&sh_uart->ctrl_blk->smp, dummy)) {
```

Comment:
This fix seems fine. However who said we have actually read some byte with:
```
rd_len = uart_fifo_read(dev, &dummy, 1);
```
?

Shouldn't we check `rd_len`?

Reply:
Yeah, this is actually good point and I have missed this one. I have been preparing fix for the SMP part which also fails to check `rd_len` and always sets `new_data` to `true`. Adding the fix here as second commit.

Reply:
I've seen your fix. Please also add to the commit message rationale why `rd_len != 0` was added and what it solves (prevents feeding smp_shell with random byte from stack).